### PR TITLE
feat: Document filters

### PR DESCRIPTION
**Pull request by mingard**

### CODE REVIEW

File: dadi/frontend/components/DocumentFilter/DocumentFilter.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      '$gt': 'Greater than',
+      '$gte': 'Greater than or Equal to',
+      '$lt': 'Less than',
+      '$lte': 'Less than or Equal to'
+    }
+  }
+
+  render() {
+    const { key, value } = this.props
+    return (
```
Comment (eduardoboucas) [inline dadi/frontend/components/DocumentFilter/DocumentFilter.jsx]: Blank line here.

File: dadi/frontend/components/DocumentFilters/DocumentFilters.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import { isValidJSON } from 'lib/util'
+
+import DocumentFilter from 'components/DocumentFilter/DocumentFilter'
+
+export default class DocumentFilters extends Component {
+
+  render() {
+    const { filter, collection } = this.props
+    let filters = isValidJSON(filter) ? JSON.parse(this.props.filter) : null
```
Comment (eduardoboucas) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: Blank line here.

File: dadi/frontend/components/DocumentFilters/DocumentFilters.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    return (
```
Comment (eduardoboucas) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: Blank line here.

File: dadi/frontend/views/DocumentList/DocumentList.jsx [hunk 4]
```diff
@@ -5,17 +5,20 @@ import { bindActionCreators } from 'redux'
...
   
   constructor (props) {
     super(props)
     this.keyboard = new Keyboard()
   }
 
+
   render() {
     const { state } = this.props
     return (
```
Comment (eduardoboucas) [inline dadi/frontend/views/DocumentList/DocumentList.jsx]: (Not part of the PR, but blank line here)

File: dadi/lib/watch/index.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+const path = require('path')
+
+const Watch = function () {
+  this.root = process.cwd()
+}
+
+Watch.prototype.getConfig = function (config) {
+  let conf = convict(config.root())
+  let env = config.get('env')
+  try {
```
Comment (eduardoboucas) [inline dadi/lib/watch/index.js]: Blank line here.

File: dadi/lib/watch/index.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    conf.loadFile(path.resolve(`${this.root}/config/config.${env}.json`))
+  } catch (e) {
+    console.log('Failed to load config, dropping to app defaults.')
+    return config
+  }
+  return conf.validate({})
```
Comment (eduardoboucas) [inline dadi/lib/watch/index.js]: Blank line here.

File: dadi/lib/watch/index.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+Watch.prototype.start = function () {
+  // Start watching directories for compile
+  return this
```
Comment (eduardoboucas) [inline dadi/lib/watch/index.js]: If I understand correctly, this isn't doing anything at the moment but will do in the future, right?

Comment (mingard) [inline dadi/lib/watch/index.js]: Correct

File: dadi/frontend/components/DocumentFilters/DocumentFilters.jsx (hunk 2 above)
Comment (mingard) [inline dadi/frontend/components/DocumentFilters/DocumentFilters.jsx]: Done

**Comment by eduardoboucas** [comment]
Looks ace. 💯
